CONTRIBUTING.md

```diff
@@ -0,0 +1,338 @@
+
+7. **Maintain compatibility with existing standards, libraries, and systems**
+   We want things we build to interoperate and be compatible with existing tools, systems, and expectations. So when writing a smart contract, or particular functionality, find out if there are implementation details that are already established that affect how the functionality works, and make sure your implementation works the way that will be expected. I'm not talking about how the code is written, but how it works, how it functions. We can write our code better (more clear, more readable, and better documented), but make it function the same as established smart contract functionality.
+
+   When implementing new functionality, here are some things you need to consider and do to ensure interoperability and to meet existing expectations of functionality:
+
+   1. Are there any [ERC standards](https://eips.ethereum.org/erc) that cover the functionality? If so, should probably follow that.
+   2. Has an existing established library such as [OpenZeppelin](https://github.com/OpenZeppelin/openzeppelin-contracts-upgradeable) already implemented that functionality in their library? Make sure your version functions the same -- emits the same events, issues the same error messages, reverts when it reverts, etc. Generally we want to match existing widespread adopted functionality. We don't want to surprise our users, unless it is a good surprise.
+   3. Are there existing widespread systems, (for example OpenSea, other NFT exchanges, and DAO and voting systems), which expect contracts to function a certain way? Match it.
+
+### Diamond Contract Development
+- Design for ERC-2535 diamond standard compliance
+- Create reusable facets for maximum onchain composability
+- Use diamond storage patterns for state management
+- Follow established ERC standards for interoperability
+
+### Security Considerations
+- All functions must be thoroughly tested
+- Gas optimization is important but security comes first
+- Follow established patterns from existing implementations
+- Consider edge cases and potential attack vectors
+
+## Testing
+
+### Testing Requirements
+Before submitting a pull request:
+- Run all tests: `forge test`
+- Update gas snapshots: `forge snapshot`
+- Ensure your code follows the banned features list
+- Test your changes thoroughly
+- Write tests for new functionality
+- Ensure test coverage is maintained or improved
+
+### Running Tests
+```bash
+# Run all tests
+forge test
+
+# Run tests with verbose output
+forge test -vvv
+
+# Run specific test file
+forge test --match-path test/ERC20.sol
+
+# Run tests with gas reporting
+forge test --gas-report
+```
+
+### Test Writing Guidelines
+- Write comprehensive tests for all new functionality
+- Test edge cases and error conditions
+- Use descriptive test names
+- Follow the existing test patterns in the codebase
+- Include gas optimization tests where relevant
+
```

Comment:
Add a note here that if you don't want to write tests for new functionality then we can find someone to work with you to write the tests. Only make a pull request with full test coverage of your code.